src/sparseml/pytorch/optim/modifier_pruning.py

```diff
@@ -77,117 +79,95 @@ def _log_sparsity(
+        update_frequency: float = -1.0,
+        min_frequency: float = -1.0,
+        log_types: Union[str, List[str]] = None,
     ):
         super().__init__(
             log_types=log_types,
             start_epoch=start_epoch,
+            min_start=min_start,
             end_epoch=end_epoch,
-            end_comparator=-1,
+            min_end=min_end,
+            end_comparator=end_comparator,
+            update_frequency=update_frequency,
+            min_frequency=min_frequency,
         )
         self._params = validate_str_iterable(
             params, "{} for params".format(self.__class__.__name__)
         )
-        self._module_masks = None  # type: ModuleParamPruningMask
-        self._analyzers = None
+        self._module_masks = None  # type: Optional[ModuleParamPruningMask]
+        self._analyzers = None  # type: Optional[List[ModulePruningAnalyzer]]
         self._last_logged_epoch = None
 
-    def __del__(self):
-        del self._module_masks
-
     def state_dict(self) -> Dict[str, Tensor]:
         """
-        :return: Dictionary to store the masks currently created by this object. The
-            mapping is param_name -> mask
+        :return: PyTorch state dictionary to store any variables from this modifier.
+            The mapping is param_name -> mask
         """
         return OrderedDict(
             zip(self._module_masks.names, self._module_masks.param_masks)
         )
 
-    def load_state_dict(self, state_dict: Dict[str, Tensor]):
+    def load_state_dict(self, state_dict: Dict[str, Tensor], strict: bool = True):
         """
         Loads the given state dict into this object's modifiers
 
         :param state_dict: dictionary object as generated by this object's state_dict
             function
-        :raises RuntimeError: If any keys in the state dict do not correspond to a valid
-            parameter in this modifier or if this modifier has not been initialized
+        :param strict: Ignored for this modifier, everything is treated as strict
+        :raises IndexError: If any keys in the state dict do not correspond to a valid
+            index for this manager and strict=True
         """
         if not self.initialized:
             raise RuntimeError("Cannot load state dict for an uninitialized modifier")
 
-        mask_names = self._module_masks.names
-        for key in state_dict:
-            if key not in mask_names:
-                raise RuntimeError(
-                    f"State dict key {key} not found in ConstantPruningModifier params"
-                )
-        for name in mask_names:
-            if name not in state_dict:
-                raise RuntimeError(
-                    f"Mask parameter name {name} not found in state dict"
-                )
+        mask_names = {key for key in self._module_masks.names}
+        state_dict_keys = {key for key in state_dict.keys()}
+        diff = mask_names.symmetric_difference(state_dict_keys)
+        if diff:
```

Comment:
should check `strict` here

Reply:
Because of the way we load the masks, strict is the only option, otherwise, we'll have to fail. I put a note in the docs for this